### ISSUE TITLE
Parametric Space Mapping

### ISSUE THREAD

**Issue by sslattery** [ECP-copa/Cajita#8]
It was pointed out that really the physical grid cell size is unneeded and could be abstracted away from the core grid and array data structures to represent general structured data. A mapping class would then be created for different physical mappings to handle different domain and structured grid types.

To do this we should remove all physical space functions from the main classes and add a physical space mapping capability that gets overloaded based on some type of grid type tag (i.e. uniform Cartesian, etc.)

**Comment by sslattery** [comment]
Also useful would be a function like:
```
template<class EntityType>
double locate( EntityType, const int i, const int j, const int k ) const;
```
which would allow one to locate a given entity in a block or globally in a grid. We will have to flesh out this interface more but the functionality is needed - especially when we generalize coordinate data from uniform grids to general Cartesian.

**Comment by sslattery** [comment]
Fixed by #13
